hilti/toolchain/src/compiler/codegen/expressions.cc

```diff
@@ -215,6 +215,13 @@ struct Visitor : hilti::visitor::PreOrder {
         result = {cxx::ID(n->id()), Side::LHS};
     }
 
+    void operator()(expression::ConditionTest* n) final {
+        auto type = cg->compile(n->type()->type()->as<type::Result>()->dereferencedType(), codegen::TypeUsage::Storage);
+        result = fmt(
+            "(%s ? ::hilti::rt::make_result(::hilti::rt::Nothing{}) : ::hilti::rt::Result<::hilti::rt::Nothing>(%s))",
```

Comment:
Nit: we want an `error` on the `else` branch and could directly get it out of `n` (`error()` already returns a correctly coerced result).

```suggestion
            "(%s ? ::hilti::rt::make_result(::hilti::rt::Nothing{}) : :%s)",
```